### ISSUE TITLE
Implement `lowerSoftmax` UOp Lowering Function

### ISSUE THREAD

**Issue by Hitomamacs** [ZantFoundation/Z-Ant#364]
# Feature Request: Implement `lowerSoftmax` UOp Lowering Function

## Summary
Implement the `lowerSoftmax` function to lower the ONNX Softmax operation into UOps for execution in the Z-Ant tensor framework.

## Description
The Z-Ant codebase currently has:
- ✅ Tensor math implementation: `op_softmax.zig` 
- ✅ IR graph operator: `op_softmax.zig` in IR_graph
- ❌ **Missing UOps lowering function: `lowerSoftmax`**

## ONNX Specification
**ONNX Softmax Documentation**: https://onnx.ai/onnx/operators/onnx__Softmax.html

**Operation**: Softmax
- **Domain**: `""`  
- **Since version**: 13
- **Function**: The operator computes the normalized exponential values for the given input: Softmax(input, axis) = Exp(input) / ReduceSum(Exp(input), axis=axis, keepdims=1)
- **Attributes**: 
  - `axis` (int, default=-1): Describes the dimension Softmax will be performed on
- **Inputs**: 
  - `input` (T): Input tensor
- **Outputs**:
  - `output` (T): Output tensor

### Function Signature
```zig
pub fn lowerSoftmax(
    b: *UOpBuilder,
    input_id: usize,
    out_shape: []const usize,
    out_dtype: DType,
    axis: i64, // Axis along which softmax is computed
) usize
```

### Implementation Approach
Softmax requires a more complex multi-pass approach compared to element-wise operations:

1. **Setup Phase**:
   - Create a VIEW for the input tensor
   - Define global output buffer with `DEFINE_GLOBAL`
   - Calculate axis-specific dimensions and strides

2. **Three-Pass Algorithm**:
   
   **Pass 1 - Find Maximum**:
   - `RANGE` over axis dimension
   - Find maximum value along the specified axis for numerical stability
   - `REDUCE_MAX` operation or manual comparison loop
   
   **Pass 2 - Compute Exponentials and Sum**:
   - `RANGE` over axis dimension  
   - Subtract max from each element (for stability)
   - Apply `EXP` UOp
   - Accumulate sum with `ADD`
   
   **Pass 3 - Normalize**:
   - `RANGE` over axis dimension
   - Divide each exponential by the sum
   - `DIV` UOp for final normalization

3. **Return**: SSA id of output buffer

### Expected UOp Sequence
```
VIEW(input) -> DEFINE_GLOBAL(output) -> 
RANGE(find_max) -> LOAD -> MAX -> ENDRANGE ->
RANGE(exp_sum) -> LOAD -> SUB -> EXP -> ADD -> ENDRANGE ->
RANGE(normalize) -> LOAD -> DIV -> STORE -> ENDRANGE
```

## Mathematical Definition
For input x and axis i:
1. x_max = max(x_i) for numerical stability
2. exp_vals = exp(x_i - x_max)  
3. sum_exp = Σ(exp_vals)
4. softmax(x_i) = exp_vals_i / sum_exp

## Files to Modify
- **Primary**: Create the `lowerSoftmax` function in the appropriate UOps lowering module
- **Integration**: Add softmax operation support to UOp processing pipeline with axis handling
- **Testing**: Verify compatibility with existing `op_softmax.zig` tensor math implementation

## Acceptance Criteria
- [ ] `lowerSoftmax` function implemented following Z-Ant UOp patterns
- [ ] Function correctly handles axis-specific softmax operations
- [ ] Numerical stability through max subtraction
- [ ] Integration with existing tensor math and IR graph implementations
- [ ] Proper memory management and SSA id handling
- [ ] Compatible with Z-Ant's broadcasting and shape handling systems
- [ ] Support for negative axis indexing

## Related Files
- `/src/Core/Tensor/TensorMath/lib_elementWise_math/op_softmax.zig` (tensor math)
- `/src/IR_graph/op_union/operators/op_softmax.zig` (IR graph)  
- `/src/Core/Tensor/TensorMath/lib_elementWise_math/op_addition.zig` (reference implementation)

## Priority
High - Softmax is essential for classification tasks, attention mechanisms, and probability distributions in neural networks.

## Implementation Notes
- Requires careful axis dimension handling
- Must implement numerical stability (max subtraction)
- More complex than element-wise operations due to reduction requirement
- Consider memory access patterns for optimal performance
